### DISCUSSION TITLE
virtualization

### DISCUSSION THREAD

**Discussion by Karpengold** [xyflow/xyflow#2703] [Q&A] [answered]
Is there any way to provide custom virtualization solution? I.e. I have the case with nodes and edges with static positions and I would like to render only nodes which are in the visible window

**Comment by moklick** [comment, marked as answer]
Did you try to use the `onlyRenderVisibleElements` prop  https://reactflow.dev/docs/api/react-flow-props/#onlyrendervisibleelements ?

**Reply by Karpengold** [reply]
Excellent! I don't know how I missed it, but thanks, this is exactly what I was looking for.
